src/scene/lightmapper.js

```diff
@@ -19,104 +19,88 @@ import { drawQuadWithShader } from '../graphics/simple-post-effect.js';
 import { RenderTarget } from '../graphics/render-target.js';
 import { Texture } from '../graphics/texture.js';
 
+import { MeshInstance } from './mesh-instance.js';
+
 import {
     BAKE_COLORDIR,
     FOG_NONE,
     LIGHTTYPE_DIRECTIONAL, LIGHTTYPE_OMNI, LIGHTTYPE_SPOT,
     PROJECTION_ORTHOGRAPHIC, PROJECTION_PERSPECTIVE,
     SHADER_FORWARDHDR,
     SHADERDEF_DIRLM, SHADERDEF_LM,
+    MASK_LIGHTMAP, MASK_BAKED,
     SHADOWUPDATE_REALTIME, SHADOWUPDATE_THISFRAME
 } from '../scene/constants.js';
 import { Camera } from '../scene/camera.js';
 import { GraphNode } from '../scene/graph-node.js';
 import { StandardMaterial } from '../scene/materials/standard-material.js';
 
-var maxSize = 2048;
-var maskBaked = 2;
-var maskLightmap = 4;
+const MAX_LIGHTMAP_SIZE = 2048;
 
-var sceneLightmaps = [];
-var sceneLightmapsNode = [];
-var lmCamera;
-var tempVec = new Vec3();
-var bounds = new BoundingBox();
-var lightBounds = new BoundingBox();
-var tempSphere = {};
+const PASS_COLOR = 0;
+const PASS_DIR = 1;
 
-var PASS_COLOR = 0;
-var PASS_DIR = 1;
+let tempVec = new Vec3();
+let tempSphere = {};
```

Comment:
```suggestion
let tempSphere = new pc.BoundingSphere();
```

Reply:
Note that you'd also need this at the top:

```javascript
import { BoundingSphere } from '../shape/bounding-sphere.js';
```

Reply:
done